R/FormulaSums.R

```diff
@@ -243,10 +264,38 @@ FormulaSums <- function(data, formula, makeNames = TRUE, crossTable = FALSE, tot
       }
     }
     if (makeModelMatrix) {
-      m <- rbind(m, fac2sparse(rg1, drop.unused.levels = FALSE)) 
+      if (viaSparseMatrix) {
+        if (is.factor(rg1)) {
+          n_j <- max(as.integer(levels(rg1)))
+          # rg1 <- as.integer(as.character(rg1))

```

Comment:
Is this comment needed?

Reply:
Whether to use as.integer(as.character(rg1)) or as.integer(rg1) is a tricy topic here. I keep the comment with extra comments :)